embedded-graphics/MIGRATING-0.5-0.6.md

```diff
@@ -0,0 +1,444 @@
+
+Additional color types can be created for targets where no suitable builtin color type exists.
+
+Creating various colors now works like this:
+
+```rust
+use embedded_graphics::{
+    pixelcolor::{BinaryColor, Rgb565, Rgb888},
+    prelude::*,
+};
+
+// Binary color (off/on)
+let on = BinaryColor::On;
+
+// Red with a small amount of green creates a deep orange colour
+let rust = Rgb565::new(0xff, 0x07, 0x00);
+
+// Use a color constant provided by the RGB color types
+let magenta = Rgb888::MAGENTA;
+```
+
+Pixels can also be converted to their underlying storage by calling `.into_storage()`:
+
+```rust
+use embedded_graphics::{
+    pixelcolor::Rgb565,
+    prelude::*,
+};
+
+let raw_value: u16 = Rgb565::MAGENTA.into_storage();
+```
+
+### Color constants
+
+Predefined color constants have been added to all grayscale and RGB colors.
+
+Usage:
+
+```rust
+use embedded_graphics::{prelude::*, pixelcolor::{Gray8, Rgb565}};
+
+let white = Gray8::WHITE;
+let cyan = Rgb565::CYAN;
+```
+
+#### Predefined colors for `Gray2`, `Gray4` and `Gray8`
+
+These are provided by the `GrayColor` trait impl.
+
+- `BLACK`
+- `WHITE`
+
+#### Predefined colors for `Rgb555` `Rgb565`, `Bgr555`, `Bgr565`, `Rgb888` and `Bgr888`
+
+These are provided by the `RgbColor` trait impl.
+
+- `BLACK`
+- `WHITE`
+- `RED`
+- `GREEN`
+- `BLUE`
+- `CYAN`
+- `MAGENTA`
+- `YELLOW`
+
+## General drawing operations
+
+Drawing operations are "reversed" in 0.6.0. Instead of calling `display.draw(thing)`, call `thing.draw(&mut display)`:
+
+```diff
+- let rect = Rectangle::new(Coord::new(50, 20), Coord::new(60, 35))
+-     .stroke(Some(5u8))
+-     .stroke_width(3)
+-     .fill(Some(10u8));
+-
+- display.draw(rect);
+
++ let style = PrimitiveStyleBuilder::new()
++     .stroke_color(Rgb565::RED)
++     .stroke_width(3)
++     .fill_color(Rgb565::GREEN)
++     .build();
++
++ Rectangle::new(Point::new(50, 20), Point::new(60, 35))
++     .into_styled(style)
++     .draw(&mut display)?;
+```
+
+Chaining now looks like this:
+
+```rust
+Circle::new(Point::new(64, 64), 64)
+    .into_styled(PrimitiveStyle::with_stroke(BinaryColor::On, 1))
+    .into_iter()
+    .chain(
+        &Line::new(Point::new(64, 64), Point::new(0, 64))
+            .into_styled(PrimitiveStyle::with_stroke(BinaryColor::On, 1)),
+    )
+    .chain(
+        &Line::new(Point::new(64, 64), Point::new(80, 80))
+            .into_styled(PrimitiveStyle::with_stroke(BinaryColor::On, 1)),
+    )
+    .chain(
+        &Text::new("Hello World!", Point::new(5, 50))
+            .into_styled(TextStyle::new(Font6x8, BinaryColor::On)),
+    )
+    .draw(&mut display);
+```
+
+Drawing operations are now fallible, with `.draw()` calls returning a `Result`. This allows for error handling if an error occurs during a drawing operation. The error type of this `Result` is dependent on the associated `Error` type as defined by the display driver's `DrawTarget` impl.
+
+## Coordinates and positioning
```

Comment:
Maybe
```suggestion
## Coordinates and sizing
```
?

Reply:
I went with `Coordinates, sizing and positioning` because why not cover all the bases ;)